src/Simplex_tree/concept/FiltrationValue.h

```diff
@@ -17,7 +17,19 @@
   * its subsimplices of same filtration value) provides an indexing scheme 
   * (see IndexingTag).
   */
-  struct FiltrationValue {
-    /** \brief Operator < is a StrictWeakOrdering. */
-    bool operator<(FiltrationValue f1, FiltrationValue f2);
-  };
+struct FiltrationValue {
+  /** \brief Operator < is a StrictWeakOrdering. */
+  bool operator<(FiltrationValue f1, FiltrationValue f2);
+
+  /** \brief For multiparameter filtrations, this methods pushes a filtration value 
+   * to the first moment, for operator< such that f < this. For instance, for a one critical filtration, with
+   *  - this = (1,2)
+   *  - x = (2,1)
+   *
+   * after calling this method, x should be equal to (2,2).
+   * This function is called when using, e.g. `make_filtration_non_decreasing`, as the filtration of a simplex
+   * has to be greater than the filtration of any of its faces.
+   * */ 
+  void push_to(const FiltrationValue f);
```

Comment:
`push_forward` ? I haven't though a lot about multicritical filtrations, but my first guess to implement this would be to push all of of the filtration values of `this` into each of the `f` ones, and then remove the redundant ones (for $\le$).
I think doing a concept is a good idea.

Reply:
Is "push" a standard term for posets? All I found is the pushout of 2 functors (some form of colimit), is that what you are using here? If the name matches a common definition in category theory, that can be a good reason to use it, but I don't know which one to look at.
(In my intuition, "push"ing one value does not really reflect the symmetry of the operation. Something like smallest_common_majorant (by analogy with GCD) would look more symmetric.)

Reply:
I'm not following a standard term for this, so something like you proposed seems great to me.
The analogy is more with the LCM instead of GCD, but "smallest_common_majorant" seems great to me.
As calling multiple times LCM gives the LCM of a family (here `make_filtration_non_decreasing`), this is very moral.